### ISSUE TITLE
14.8.4 breaks when using scss interpolation.

### ISSUE THREAD

**Issue by siiron** [stylelint/stylelint#6113]
### What steps are needed to reproduce the bug?

"stylelint": "14.8.4",
"stylelint-config-prettier": "9.0.3",
"stylelint-config-sass-guidelines": "9.0.1",
"stylelint-prettier": "2.0.0",

Try to lint scss-code using interpolation where parts of expression is identical as another, but not the whole expression.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-sass-guidelines",
    "stylelint-config-prettier",
    "stylelint-prettier/recommended"
  ],
  "rules": {
    "at-rule-name-space-after": "always",
    "block-no-empty": true,
    "block-opening-brace-space-before": "always",
    "color-hex-length": "long",
    "color-no-invalid-hex": true,
    "comment-no-empty": true,
    "declaration-block-no-duplicate-properties": [
      true,
      {
        "ignore": ["consecutive-duplicates-with-different-values"]
      }
    ],
    "declaration-block-no-shorthand-property-overrides": true,
    "font-family-no-duplicate-names": true,
    "font-family-no-missing-generic-family-keyword": true,
    "function-calc-no-unspaced-operator": true,
    "function-linear-gradient-no-nonstandard-direction": true,
    "keyframe-declaration-no-important": true,
    "max-nesting-depth": 3,
    "media-feature-name-no-unknown": true,
    "no-duplicate-at-import-rules": true,
    "no-duplicate-selectors": true,
    "no-empty-source": true,
    "no-extra-semicolons": true,
    "no-invalid-double-slash-comments": true,
    "order/order": [
      [
        "custom-properties",
        "dollar-variables",
        {
          "type": "at-rule",
          "name": "extend"
        },
        {
          "type": "at-rule",
          "name": "include",
          "hasBlock": false
        },
        "declarations",
        "rules"
      ]
    ],
    "property-no-unknown": true,
    "selector-class-pattern": [
      "^((hb-)([a-z]?[a-zA-Z0-9]+)*(-[a-z0-9]?[a-zA-Z0-9]+)*?(--[a-zA-Z0-9]+)*?)$",
      {
        "message": "Please make sure the selector follows the following pattern: https://regexr.com/5ca1a"
      }
    ],
    "scss/at-mixin-pattern": "^[a-z][a-zA-Z0-9]+$",
    "scss/dollar-variable-pattern": "^[_]?[a-zA-Z]+([a-zA-Z0-9-]+[a-zA-Z0-9]+)?$",
    "selector-max-compound-selectors": 3,
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-no-unknown": true,
    "selector-type-no-unknown": true,
    "string-no-newline": true,
    "unit-no-unknown": true
  }
}

```

### How did you run Stylelint?

npx stylelint "path/to/styles/**/*.scss"

### Which version of Stylelint are you using?

14.8.4

### What did you expect to happen?

No problems during linting.

### What actually happened?

Stylelint reports multiple "Unexpected duplicate selector "", first used at line XX no-duplicate-selectors"

I've traced it down to places where I use scss interpolation.

```scss
.hb-buttongroup--unstack\@#{$breakpoint-sm} {
...
}

.hb-buttongroup--unstack\@#{$breakpoint-md} {
...
}

.hb-buttongroup--unstack\@#{$breakpoint-lg} {
....
}
```

Here I get the error where I use #{$breakpoint-md} and #{$breakpoint-lg} but I think it triggers on the `hb-buttongroup--unstack\@` part

Another case

```scss
@for $i from 1 through ($number) {
    $decimal: math.div($i, $number);
    .hb-cell--#{$i}of#{$number}#{$suffiks} {
      @include elWidth($decimal);
    }

    .hb-cell--push-#{$i}of#{$number}#{$suffiks} {
      margin-left: ($decimal * 100) * 1%;
    }

    .hb-cell--pull-#{$i}of#{$number}#{$suffiks} {
      margin-left: -($decimal * 100) * 1%;
    }
}
```

Here I get the error where I use #{$i}of#{$number}#{$suffiks} even though the first part is different.

### Does the bug relate to non-standard syntax?

Yes It's related to Scss Interpolation

**Comment by boris-petrov** [comment]
The same happens with:

```scss
%foo {
  ...
}

%bar {
  ...
}
```

Errs on `%bar` and all `%XXX` after that.

**Comment by FreekVR** [comment]
I'm seeing a similar thing with `less` on this specific stylelint version, I pinned to `14.8.3` and no issues with the linter there.

![image](https://user-images.githubusercontent.com/417416/170213223-8804494e-19ae-498c-b2b7-88aaf0f092ac.png)

**Comment by ybiquitous** [comment]
Hi there, sorry for the inconvenience. 

This regression has been fixed but not released yet (duplicate of #6110). I'll release the fixed version today or tomorrow.

**Comment by ybiquitous** [comment]
[14.8.5](https://github.com/stylelint/stylelint/releases/tag/14.8.5) released! 🎉
